app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,37 @@
+class MoviesController < ApplicationController
+  def index
+    movies = Movie.all
+    render json: jsonify(movies)
+  end
+
+  def show
+    movie = Movie.find_by(id: params[:id])
+    if movie
+      render json: jsonify(movie)
+    else
+      # head :not_found
+      render json: { errors: { movie_id: ["No such movie"] }}, status: :not_found
+      # render json: {}, status: :not_found
+    end
+  end
+
+  def create
+    movie = Movie.new(movie_params)
+    movie.available_inventory = movie_params[:inventory]
+    if movie.save
+      render json: {id: movie.id}
+    else
+      render_error(:bad_request, movie.errors.messages)
+    end
+  end
+
+  private
+
+  def movie_params
+    params.require(:movie).permit(:title, :overview, :release_date, :inventory)
```

Comment:
In Rails when we used forms, `params` would come back populated with a nested structure that had `movie` in it. In this API, we won't have that nested structure, so we need to take out the `.require('movie')` bit. When we do this, then the rest of Wave 2 starts working :)